### ISSUE TITLE
Dynamic service routing not working with Eureka

### ISSUE THREAD

**Issue by macsux** [ThreeMammals/Ocelot#846]
## Expected Behavior / New Feature
Configuring Ocelot for dynamic routing as per https://ocelot.readthedocs.io/en/latest/features/routing.html#dynamic-routing with Eureka should allow routing to /SERVICE-NAME/*

## Actual Behavior / Motivation for New Feature
Unable to find the route
Service Discovery works if service is configured explicitly as a route (non-dynamic)

## Steps to Reproduce the Problem
Sample project showing the issue is provided:
1. Clone https://github.com/macsux/dotnet-ers-refarch
2. Start eureka service (docker-compose provided in /dependencies)
3. Run `web-bff-tracing` project. Confirm that it registered in Eureka by accessing eureka dashboard (http://localhost:8761/)
4. Run `gateway` project
5. Attempt to access http://localhost:5002/WEB-BFF-TRACING

## Version
Ocelot: 13.4.1
ASP.NET Core 2.2

**Comment by thiagoloureiro** [comment]
We had an similar issue:
https://github.com/ThreeMammals/Ocelot/issues/655

Can you confirm if this is something related? then will be easier to investigate. thanks!

**Comment by thiagoloureiro** [comment]
![image](https://user-images.githubusercontent.com/29433794/55323939-d5248c00-5480-11e9-8b91-e14fa35b97f8.png)

I want to understand why this /x/  ? (attached)

I removed the x and worked ok!

![image](https://user-images.githubusercontent.com/29433794/55323951-e2da1180-5480-11e9-8e5c-9690c2e31313.png)

More info:

![image](https://user-images.githubusercontent.com/29433794/55324135-76134700-5481-11e9-8262-22b8dad54fbb.png)

ocelot.json

```
{
  "ReRoutes": [
    {
      "DownstreamPathTemplate": "/{url}",
      "DownstreamScheme": "http",
      "UpstreamPathTemplate": "/WEB-BFF-TRACING/{url}",
      "UpstreamHttpMethod": [ "Get" ],
      "ServiceName": "WEB-BFF-TRACING",
      "LoadBalancerOptions": {
        "Type": "LeastConnection"
      }
    }
  ],

  "GlobalConfiguration": {
    "ServiceDiscoveryProvider": {
      "Type": "Eureka"
    },
    "DownstreamScheme": "http"
  }
}

```

**Comment by macsux** [comment]
That's not dynamic routing. Clear out the routes section (I updated the sample so it's empty). The routes should be coming from eureka - I should not have to declare them at all. This is how the config should look look like according to docs for dynamic routes to work:

```
{
ReRoutes: [ ],
GlobalConfiguration: {
ServiceDiscoveryProvider: {
Type: "Eureka"
},
DownstreamScheme: "http"
}
}
```

**Comment by thiagoloureiro** [comment]
@macsux  let me check now :) will get back to you soon

**Comment by macsux** [comment]
Any update on this?

**Comment by raman-m** [comment]
@macsux  Hi Andrew!
How things going?
Yes, seems Dynamic Routing feature works with Consul only being designed by Tom many years ago. 😄 
It is expected fact that for other types of service discovery providers it doesn't work.
But I believe design & implementation of SD+Eureka+DynRouting triple will not be so complex, but yes, a some refactoring should be done. Probably a couple of adapters & bridges should help.

Do you use Ocelot nowadays? If Yes, please upgrade your solutions to v22!

We are ready to accept this feature request if you have intention to contribute. If Not, I have no idea when we will start...
I'll keep this issue open...

**Comment by macsux** [comment]
I'm no longer interested in ocelot now that yarp is a thing. Thanks

**Comment by raman-m** [comment]
@macsux 🆗 Adrew! With high probability we'll close the ticket...

@RaynaldM @ggnaegi 
The author is not interested in development of this feature. My concern here is that Eureka is not so popular Service Discovery provider as Consul one...
**Does it make sense to keep the issue open for contributions by the community?
And how to check the popularity of Eureka service?**

**Comment by ggnaegi** [comment]
@raman-m I would close the issue, most of the issues are related to consul not eureka. It means to me, that most of the users are working with consul not eureka. Besides, they can add their own providers if they want. With yarp, no other choice but to implement your own provider (updating the configuration yourself, routes and clusters).

**Comment by RaynaldM** [comment]
@raman-m it's probably time to cut back on compatibility with certain third-party services and concentrate on the ones we use most.
If anyone is interested in developing a particular service, they can do what we're doing: play an active part in Ocelot's development.

**Comment by ggnaegi** [comment]
@RaynaldM @raman-m I agree with Raynald, we should concentrate ourselves on the providers we know the community is using. We just can't support all kinds of providers.

**Comment by raman-m** [comment]
👌 We've reached team consensus.
Closing...

**Comment by macsux** [comment]
Guys, to be frank it's been almost 5 years since that issue was filled by
me. It's just not realistic for someone to wait around that long without
moving on to alternatives. Not sure what prompted revived interest, but it
seemed like ocelot was dead project for a while with seldom any commits.

**Comment by ggnaegi** [comment]
@macsux Or you do like some of us, you fork the solution and refactor it for your needs 😸 Yes, project was dead for a while - and that's why from my side I modified Ocelot for our needs -. Some decided to relaunch ocelot and propose an independent solution. Sure, it's far from perfect, but it's improving and it's 100% open source.
